configure.ml

```diff
@@ -235,6 +235,101 @@ let _ = if not (dir_exists "bin") then Unix.mkdir "bin" 0o755
 
 type ide = Opt | Byte | No
 
+type preferences = {
+  prefix : string option;
+  local : bool;
+  vmbyteflags : string option;
+  custom : bool option;
+  bindir : string option;
+  libdir : string option;
+  configdir : string option;
+  datadir : string option;
+  mandir : string option;
+  docdir : string option;
+  emacslib : string option;
+  coqdocdir : string option;
+  ocamlfindcmd : string option;
+  lablgtkdir : string option;
+  camlp5dir : string option;
+  arch : string option;
+  natdynlink : bool;
+  coqide : ide option;
+  macintegration : bool;
+  browser : string option;
+  withdoc : bool;
+  byteonly : bool;
+  flambda_flags : string list;
+  debug : bool;
+  profile : bool;
+  bin_annot : bool;
+  annot : bool;
+  bytecodecompiler : bool;
+  nativecompiler : bool;
+  coqwebsite : string;
+  force_caml_version : bool;
+  force_findlib_version : bool;
+  warn_error : bool;
+}
+
+module Profiles = struct
+
+let default = {
+  prefix = None;
+  local = false;
+  vmbyteflags = None;
+  custom = None;
+  bindir = None;
+  libdir = None;
+  configdir = None;
+  datadir = None;
+  mandir = None;
+  docdir = None;
+  emacslib = None;
+  coqdocdir = None;
+  ocamlfindcmd = None;
+  lablgtkdir = None;
+  camlp5dir = None;
+  arch = None;
+  natdynlink = true;
+  coqide = None;
+  macintegration = true;
+  browser = None;
+  withdoc = false;
+  byteonly = false;
+  flambda_flags = [];
+  debug = true;
+  profile = false;
+  bin_annot = false;
+  annot = false;
+  bytecodecompiler = true;
+  nativecompiler = not (os_type_win32 || os_type_cygwin);
+  coqwebsite = "http://coq.inria.fr/";
+  force_caml_version = false;
+  force_findlib_version = false;
+  warn_error = false;
+}
+
+let devel = { default with
+  local = true;
+  bin_annot = true;
+  annot = true;
+  warn_error = true;
+}
+let devel_doc = "-local -annot -bin-annot -warn-error yes"
+
+let get = function
+  | "devel" -> devel
+  | s -> raise (Arg.Bad ("profile name expected instead of "^s))
+
+let doc =
+  "<profile> Sets a bunch of flags. Supported profiles:
+     devel = " ^ devel_doc
```

Comment:
A more generic way to do this would be to have an associative list of `profile key * (profile value * profile doc)`.

Reply:
I agree, see also my comment below.